### PR TITLE
maint(test): make sure sphinx warnings fail on actions

### PR DESCRIPTION
**Pull request by oscarbenjamin**
#### References to other Issues or PRs

See https://github.com/sympy/sympy/pull/19867#issuecomment-743098286

#### Brief description of what is fixed or changed

Use `set -o errexit` in `test_sphinx.sh` to fail the test on actions if the build fails.

#### Other comments

I've added a second commit that I will remove later that is designed to fail the sphinx build so we can see that actions reports the failure.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Looks like this works and fails the sphinx build on actions as hoped.

I'll remove the broken commit to make this mergeable.

**Review by eric-wieser — APPROVED** [review]

**Comment by oscarbenjamin** [comment]
I've added a bunch more commits to finish migrating to Actions (#20374). This is close but still needs the Python 2 import test.

File: bin/test_sphinx.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #! /usr/bin/env bash
 
+set -o errexit
```
Comment (asmeurer) [inline bin/test_sphinx.sh:3]: We should make sure all the bash scripts use this. Also `set -x` is useful (it prints what commands are being run).

Comment (oscarbenjamin) [reply bin/test_sphinx.sh:3]: The fact that sane error handling is an opt-in feature in bash never ceases to amaze me.

**Comment by oscarbenjamin** [comment]
This is looking good enough to disable Travis. I'm going to merge this without waiting for Travis to finish.

**Pull request merged**